### PR TITLE
Allow configuration of a self-hosted Plausible instance

### PR DESCRIPTION
**Pull request by luizkowalski**
hey 👋🏻 

I've been playing around with self-hosting my Plausible instance and notice that the gem defaults to `plausible.io`

I quickly put together an option to customize it but defaults to `plausible.io`

### CODE REVIEW

**Comment by luctus** [comment]
@luizkowalski hey!

Thanks a lot for your contribution! I will add some tests and publish it as soon as possible.

In a RubyonRails app, you should put this code in an initiliazer, right?

```rb
PlausibleApi.configure do |config|
  config.base_url = "https://your-plausible-instance.com/"
end
```

File: lib/plausible_api/configuration.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module PlausibleApi
+  class Configuration
+    attr_accessor :base_url
+
+    def base_url
+      @base_url || "https://plausible.io"
```
Comment (jnunemaker) [inline lib/plausible_api/configuration.rb]: What I often do is setup defaults in initialize and then do an attr_accessor as you have for overriding. Then you aren't making the same string over and over. Minor so every one feel free to ignore me.

**Comment by jnunemaker** [comment]
> In a RubyonRails app, you should put this code in an initializer, right?

@luctus as a rails user, yep! That's what we recommend for flipper.

File: lib/plausible_api/configuration.rb (hunk 1 above)
Comment (luizkowalski) [reply lib/plausible_api/configuration.rb]: updated. let me know if that's what you meant.

**Pull request merged**